pkg/resource/training_job/hooks.go

```diff
@@ -64,4 +69,23 @@ func (rm *resourceManager) customSetOutput(r *resource) {
 	}
 
 	svccommon.SetSyncedCondition(r, trainingJobStatus, &resourceName, &trainingJobModifyingStatuses)
+
+	warmpoolUsed := ackcompare.IsNotNil(r.ko.Spec.ResourceConfig) && ackcompare.IsNotNil(r.ko.Spec.ResourceConfig.KeepAlivePeriodInSeconds)
+
+	// Only requeue when warm pool is being used and when training job is in the completed state.
+	// WP will always have terminated status on error(Training Job or Warmpool).
+	if ackcompare.IsNotNil(trainingJobStatus) && *trainingJobStatus == svcsdk.TrainingJobStatusCompleted &&
+		warmpoolUsed {
+
+		// Sometimes DescribeTrainingJob does not contain the warm pool status
+		// In this condition the only possible status is Available or Terminated.
+		if ackcompare.IsNotNil(trainingJobStatus) && ackcompare.IsNil(r.ko.Status.WarmPoolStatus) {
```

Comment:
checking if trainingJobStatus is not nil again?

Reply:
Nice catch, I made in the change in the update training job pr.